zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbResourceState.java

```diff
@@ -291,6 +279,7 @@ public void clearCache() {
   private PersistedResource getPersistedResourceById(
       final String resourceId, final String tenantId) {
     dbResourceId.wrapString(resourceId);
+    tenantIdKey.wrapString(tenantId);
```

Comment:
:grimacing: That sort of bug is already a security issue, right? Such an easy mistake to make with pretty bad consequences.

Reply:
We wrap it in the `getLatestResourceVersion` method, so we didn't have a bug, but it's always preferable to set it explicitly since we may refactor the code and miss it in the future.